elements/rh-dialog/rh-dialog.ts

```diff
@@ -253,6 +260,11 @@ export class RhDialog extends LitElement {
     }
   }
 
+  #onNativeDialogCancel(event: Event) {
```

Comment:
Canyou add tests for this, ensuring g that our behaviour matches the native dialog apis? 

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/dialog#handling_the_return_value_from_the_dialog